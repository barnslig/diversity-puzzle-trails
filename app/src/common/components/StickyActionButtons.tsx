import { Box } from "@mui/material";
import * as React from "react";

type StickyActionButtonsProps = {
  children: React.ReactNode;
};

/**
 * A container for fullWidth action buttons that makes them sticky to the bottom
 */
const StickyActionButtons = ({ children }: StickyActionButtonsProps) => (
  <Box position="sticky" bottom={0} paddingY={2} bgcolor="background.default">
    {children}
  </Box>
);

export default StickyActionButtons;
