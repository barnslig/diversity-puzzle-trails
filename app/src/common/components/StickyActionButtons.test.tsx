import * as React from "react";
import render from "../testing/render";

import StickyActionButtons from "./StickyActionButtons";

it("renders correctly", () => {
  const { container } = render(
    <StickyActionButtons>
      <div />
    </StickyActionButtons>
  );
  expect(container.firstChild).toMatchSnapshot();
});
