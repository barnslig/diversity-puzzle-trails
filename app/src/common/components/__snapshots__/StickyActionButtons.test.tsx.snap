// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
<div
  class="MuiBox-root css-7fvgf6"
>
  <div />
</div>
`;
